lib/http1.c

```diff
@@ -540,7 +545,7 @@ static void handle_incoming_request(struct st_h2o_http1_conn_t *conn)
             if (create_entity_reader(conn, headers + entity_body_header_index) != 0) {
                 return;
             }
-            conn->req.write_req.cb = h2o_write_req_first;
+            conn->req.write_req.cb = should_upgrade_to_http2(conn) ? write_req_non_streaming : h2o_write_req_first;
```

Comment:
I think this is controversial. This current implementation prefers upgrade to body streaming, but it might be better to rather select body streaming on h1 regarding the performance. What do you think @kazuho ?